service/history/api/signalwithstartworkflow/convert.go

```diff
@@ -58,6 +58,7 @@ func ConvertToStartRequest(
 		SearchAttributes:         request.GetSearchAttributes(),
 		Header:                   request.GetHeader(),
 		WorkflowStartDelay:       request.GetWorkflowStartDelay(),
+		UserMetadata:             request.UserMetadata,
```

Comment:
Do expect to have any kind of frontend validation for the data? For instance, that it's valid UTF-8? We've had some issues with that in the recent past (happy to elaborate).

Reply:
Yes. Both fields in the UserMetadata is `Payload` type which is expected to be binary data.